atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepableCells.java

```diff
@@ -0,0 +1,251 @@
+        SweepableCellsTable.SweepableCellsColumnValue colVal = createColVal(info.timestamp(), index, writeRef);
+        cells.put(SweepQueueUtils.toCell(row, colVal), colVal.persistValue());
+    }
+
+    private long numberDedicatedRows(List<WriteInfo> writes) {
+        return 1 + (writes.size() - 1) / MAX_CELLS_DEDICATED;
+    }
+
+    private static long tsMod(long timestamp) {
+        return timestamp % SweepQueueUtils.TS_FINE_GRANULARITY;
+    }
+
+    private SweepableCellsTable.SweepableCellsColumnValue createColVal(long ts, long index, WriteReference writeRef) {
+        SweepableCellsTable.SweepableCellsColumn col = SweepableCellsTable.SweepableCellsColumn.of(tsMod(ts), index);
+        return SweepableCellsTable.SweepableCellsColumnValue.of(col, writeRef);
+    }
+
+    private void populateWrites(SweepableCellsTable.SweepableCellsRow row, Map.Entry<Cell, Value> entry,
+            Map<CellReference, WriteInfo> writes) {
+        SweepableCellsTable.SweepableCellsColumn col = computeColumn(entry);
+
+        if (isReferenceToDedicatedRows(col)) {
+            addWritesFromDedicated(row, col, writes);
+        } else {
+            addWriteFromValue(getTimestamp(row, col), entry.getValue(), writes);
+        }
+    }
+
+    private boolean isReferenceToDedicatedRows(SweepableCellsTable.SweepableCellsColumn col) {
+        return col.getWriteIndex() < 0;
+    }
+
+    private void addWritesFromDedicated(SweepableCellsTable.SweepableCellsRow row,
+            SweepableCellsTable.SweepableCellsColumn col, Map<CellReference, WriteInfo> writes) {
+        List<byte[]> dedicatedRows = computeDedicatedRows(row, col);
+        RowColumnRangeIterator iterator = getColumnRangeAll(dedicatedRows);
+        iterator.forEachRemaining(entry -> addWriteFromValue(getTimestamp(row, col), entry.getValue(), writes));
+    }
+
+    private void addWriteFromValue(long timestamp, Value value, Map<CellReference, WriteInfo> writes) {
+        WriteReference writeRef = SweepableCellsTable.SweepableCellsColumnValue.hydrateValue(value.getContents());
+        updateLatestForCell(timestamp, writeRef, writes);
+    }
+
+    private void updateLatestForCell(long ts, WriteReference writeRef, Map<CellReference, WriteInfo> writes) {
+        WriteInfo newWrite = WriteInfo.of(writeRef, ts);
+        writes.merge(writeRef.cellReference(), newWrite,
+                (writeOne, writeTwo) -> writeOne.timestamp() > writeTwo.timestamp() ? writeOne : writeTwo);
+    }
+
+    private List<RangeRequest> rangeRequestsDedicatedRows(ShardAndStrategy shardAndStrategy, long partitionFine) {
+        SweepableCellsTable.SweepableCellsRow startingRow = computeRow(partitionFine, shardAndStrategy);
+        RowColumnRangeIterator rowIterator = getColumnRangeAllForRow(startingRow);
+        List<RangeRequest> requests = new ArrayList<>();
+        rowIterator.forEachRemaining(entry -> addRangeRequestIfDedicated(startingRow, computeColumn(entry), requests));
```

Comment:
Can we rework this to filter the rows and then add? This would avoid having to pass around another output parameter.